tracer/test/Datadog.Trace.TestHelpers/NativeProcess/SuspendedProcess.cs

```diff
@@ -0,0 +1,91 @@
+// <copyright file="SuspendedProcess.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Datadog.Trace.TestHelpers.NativeProcess;
+
+internal class SuspendedProcess(int pid, SafeProcessHandle processHandle, SafeHandle threadHandle, StreamWriter? standardInput, StreamReader? standardOutput, StreamReader? standardError) : IDisposable
+{
+#if NETFRAMEWORK
+    private const string StandardInputFieldName = "standardInput";
+    private const string StandardOutputFieldName = "standardOutput";
+    private const string StandardErrorFieldName = "standardError";
+#else
+    private const string StandardInputFieldName = "_standardInput";
+    private const string StandardOutputFieldName = "_standardOutput";
+    private const string StandardErrorFieldName = "_standardError";
+#endif
+
+    private readonly SafeProcessHandle _processHandle = processHandle;
+    private readonly SafeHandle _threadHandle = threadHandle;
+
+    private readonly StreamWriter? _standardInput = standardInput;
+    private readonly StreamReader? _standardOutput = standardOutput;
+    private readonly StreamReader? _standardError = standardError;
+
+    private bool _isResumed;
+    private bool _isDisposed;
+
+    public int Id { get; } = pid;
+
+    public Process ResumeProcess()
+    {
+        if (_isDisposed)
+        {
+            throw new ObjectDisposedException(nameof(SuspendedProcess));
+        }
+
+        _isResumed = true;
```

Comment:
Should we check against resuming twice? do we care?

Reply:
Meh. We're grown-ups.